modules/consensus/accept_test.go

```diff
@@ -701,3 +701,66 @@ func TestTaxHardfork(t *testing.T) {
 		t.Fatal("siafund pool was not increased correctly")
 	}
 }
+
+// mockGatewayDoesBroadcast implements modules.Gateway to mock the Broadcast
+// method.
+type mockGatewayDoesBroadcast struct {
+	modules.Gateway
+	broadcastCalled chan struct{}
+}
+
+// Broadcast is a mock implementation of modules.Gateway.Broadcast that
+// sends a sentinel value down a channel to signal it's been called.
+func (g *mockGatewayDoesBroadcast) Broadcast(string, interface{}) {
+	g.broadcastCalled <- struct{}{}
+}
+
+// TestAcceptBlockBroadcasts tests that AcceptBlock broadcasts valid blocks and
+// that managedAcceptBlock does not.
+func TestAcceptBlockBroadcasts(t *testing.T) {
+	cst, err := blankConsensusSetTester("TestAcceptBlockBroadcasts")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer cst.Close()
+	mg := &mockGatewayDoesBroadcast{
+		Gateway:         cst.cs.gateway,
+		broadcastCalled: make(chan struct{}),
+	}
+	cst.cs.gateway = mg
```

Comment:
You create a full object, and then replace the gateway of the full object with the mocked object. That to me says that there are going to be problems somewhere, because there's a lot of stuff happening under the hood where objects and whatnot could be getting passed around. So, the non-mocked gateway could be leaking into the consensus set and into the other parts of the consensus set tester. For example, the transaction pool is using a full gateway, which has the potential to cause problems.